src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/PageActionInvoker.cs

```diff
@@ -278,6 +277,12 @@ private Task BindArgumentsAsync()
             {
                 _result = new PageResult();
             }
+
+            // We also have some special initialization we need to do for PageResult.
+            if (_result is PageResult pageResult)
+            {
+                pageResult.ViewData = pageResult.ViewData ?? _pageContext.ViewData;
```

Comment:
What the _pageContext.ViewData will have in it? What if it's still null?

Reply:
Could you maybe leave a comment that says this is to preserve compat? 

@mkArtakMSFT for most parts, `PageResult.ViewData` is null (e.g. when you do `return Page();`). However, a user could new up an instance and explicitly assign it. That said, all this PR is doing is reviving some code we moved but didn't realize broke identity when we made the change: https://github.com/aspnet/Mvc/pull/7639/files#diff-d8384a52abde5770b0be9183b6b6c294L274

Reply:
It would be empty if the user hasn't set anything, for example in a handler method. I don't think its ever null at this point unless the user has explicitly done so. If a user does it (although not sure what the scenario would be), there could be failures in several places in our code.